metabase/metabase
# Fix stacked redirects to homepage dashboard crashing the app

Closes #43800

Avoids navigating a user to `/` and then back to the homepage dashboard while being on the homepage dashboard. Clicking the logo multiple times in a row would stack up a few redirects, each of them canceling the previous one and eventually leading to an error

I don't feel great about duplicating this logic in both the logo and the sidebar's Home link. I've tried solving this closer to the [redirect](https://github.com/metabase/metabase/blob/dfdaab03b3200cc21aa0ea3e4b6f8b0393090038/frontend/src/metabase/home/components/HomePage/HomePage.tsx#L101) itself, but we don't seem to have a clean way to access the previous route while inside that hook to disable navigation. For the sake of fixing a P1 this feels alright though, looking forward to feedback and suggestions 🙏 

### To verify

1. Go to `/admin/settings/general#custom-homepage` and turn any dashboard into the homepage
2. Open `/`
3. Click the Metabase logo at the top left multiple times
4. Click the "Home" link in the navigation sidebar multiple times 
5. Ensure you stay on the homepage and there are no reloads or errors

### Demo

**Before**

https://github.com/metabase/metabase/assets/17258145/16ea07f0-24aa-4f2f-832b-d44058615f43

**After**

https://github.com/metabase/metabase/assets/17258145/1171814c-107b-476c-ab2a-b1db4cfc5ef1
